File: src/jvmMain/java/com/dropbear/components/EntityTransformNative.java
package com.dropbear.components;

import com.dropbear.EucalyptusCoreLoader;
import com.dropbear.math.Transform;

public class EntityTransformNative {
    static {
        new EucalyptusCoreLoader().ensureLoaded();
    }

    public static native boolean entityTransformExistsForEntity(long worldPtr, long entityId);

    public static native Transform getLocalTransform(long worldPtr, long entityId);
    public static native void setLocalTransform(long worldPtr, long entityId, Transform transform);
    public static native Transform getWorldTransform(long worldPtr, long entityId);
    public static native void setWorldTransform(long worldPtr, long entityId, Transform transform);
    public static native Transform propagateTransform(long worldPtr, long entityId);
}

File: src/nativeMain/kotlin/com/dropbear/components/EntityTransform.native.kt
package com.dropbear.components

import com.dropbear.EntityId
import com.dropbear.math.Transform

internal actual fun EntityTransform.getLocalTransform(entityId: EntityId): Transform {
    TODO("Not yet implemented")
}

internal actual fun EntityTransform.setLocalTransform(
    entityId: EntityId,
    transform: Transform
) {
}

internal actual fun EntityTransform.getWorldTransform(entityId: EntityId): Transform {
    TODO("Not yet implemented")
}

internal actual fun EntityTransform.setWorldTransform(
    entityId: EntityId,
    transform: Transform
) {
}

internal actual fun EntityTransform.propagateTransform(entityId: EntityId): Transform? {
    TODO("Not yet implemented")
}

internal actual fun entityTransformExistsForEntity(entityId: EntityId): Boolean {
    TODO("Not yet implemented")
}